fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -323,6 +339,16 @@ def __init__(
 
         self._has_params = len(params) > 0
 
+        # TODO(anj): Should we conditionally do this only if we have params?
+        # TODO(anj): this won't work for MOE params.
+        self.buffer_size = sum(p.numel() for p in params)
+        self.ssd_buffer_filename = ""
+        if self.ssd_offload:
+            self.ssd_buffer_filename = f"{randint(1, int(10E6))}_rank{self.rank}"
+            self.ssd_buffer = ssd_offload.SsdBuffer(self.buffer_size, self.ssd_buffer_filename)
+            self.move_grads_to_cpu = True
+            self.move_params_to_cpu = True
```

Comment:
perhaps asserting these are set correctly by the user instead of changing them on the fly? this way users won't be surprised when they find out these are changed.

Reply:
That makes sense but what would they set this to with ssd_offload. The default is False and the docs should mention that ssd_offload will imply movement of grads and params to CPU.